### PR TITLE
[Testing] Updated tests to better handle resources that don't have standard outputs + updated 'skip' behavior + fixed Pester test result reference + fixed readme param name

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Fixed incorrect header test (fails all child modules)
- Updated the tests to check if a primary resource exists in the module and only then test for standard outputs. A primary resource is defined by a resource that matches the module path. For example: The SecurityCenter module is in a path `Microsoft.Security\azureSecurityCenter` even though this resource type does not exist in the module. Instead the module is a collection of other resource type deployments and as a result it does not make sense to test for e.g. a `Microsoft.Security\azureSecurityCenter` resource ID
- Fixed the reference of Pester test results. Currently shows 0 tests in the pipeline. With the change it shows:
  image
- Also, I updated how tests are skipped using the Pester specific 'Skip' function. This allows us to be more specific in our test outputs and tests won't just show 'green' even though they didn't actually run if skipped. The result looks like this:
  image

| Pipeline |
| - |
| [AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml) |
| [Security: AzureSecurityCenter](https://github.com/Azure/ResourceModules/actions/workflows/ms.security.azuresecuritycenter.yml) |
| [Authorization: PolicySetDefinitions](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policysetdefinitions.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: arm/.global/global.module.tests.ps1 [hunk 1]
```diff
@@ -117,7 +117,7 @@ Describe 'File/folder tests' -Tag Modules {
         It '[<moduleFolderName>] folder should contain one or more *parameters.json files' -TestCases $folderTestCases {
 
             param(
-                $moduleFolderName,
+                [string] $moduleFolderName,
                 $moduleFolderPath
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/.global/global.module.tests.ps1:121]: We can.

File: .github/actions/templates/validateModulePester/action.yml [hunk 2]
```diff
@@ -172,6 +164,6 @@ runs:
 
     - name: 'Publish Test Results'
       uses: EnricoMi/publish-unit-test-result-action@v1
-      if: always() && !contains('cancelled,skipped', steps.pester_api_run_step.outcome)
+      if: always()
```
Comment (eriqua) [inline .github/actions/templates/validateModulePester/action.yml:167]: Just checking, any alignment needed for ADO or not applicable?

File: arm/.global/global.module.tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/.global/global.module.tests.ps1:121]: done

File: .github/actions/templates/validateModulePester/action.yml (hunk 2 above)
Comment (MrMCake) [reply .github/actions/templates/validateModulePester/action.yml:167]: No. The ADO publish task is the default native - that works like a charm.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**